src/benchmarklib/synthetic_table_generator.cpp

```diff
@@ -0,0 +1,200 @@
+    auto allocator_chunk = get_allocator_for_type<Chunk>(node_id, numa_distribute_chunks);
+
+    auto segments = Segments(allocator_ptr_base_segment);
+
+    for (auto column_index = ColumnID{0}; column_index < num_columns; ++column_index) {
+      node_id = (node_id + 1) % Topology::get().nodes().size();
+
+      resolve_data_type(column_data_types[column_index], [&](const auto column_data_type) {
+        using ColumnDataType = typename decltype(column_data_type)::type;
+
+        // get typed allocators
+        auto allocator_value_segment =
+            get_allocator_for_type<ValueSegment<ColumnDataType>>(node_id, numa_distribute_chunks);
+        auto allocator = get_allocator_for_type<ColumnDataType>(node_id, numa_distribute_chunks);
+
+        std::vector<int> values;
+        values.reserve(chunk_size);
+        const auto& column_data_distribution = column_data_distributions[column_index];
+
+        // generate distribution from column configuration
+        switch (column_data_distribution.distribution_type) {
+          case DataDistributionType::Uniform: {
+            auto uniform_dist = boost::math::uniform_distribution<double>{column_data_distribution.min_value,
+                                                                          column_data_distribution.max_value};
+            generate_value_by_distribution_type = [uniform_dist, &probability_dist, &pseudorandom_engine]() {
+              const auto probability = probability_dist(pseudorandom_engine);
+              return static_cast<int>(std::floor(boost::math::quantile(uniform_dist, probability)));
+            };
+            break;
+          }
+          case DataDistributionType::NormalSkewed: {
+            auto skew_dist = boost::math::skew_normal_distribution<double>{column_data_distribution.skew_location,
+                                                                           column_data_distribution.skew_scale,
+                                                                           column_data_distribution.skew_shape};
+            generate_value_by_distribution_type = [skew_dist, &probability_dist, &pseudorandom_engine]() {
+              const auto probability = probability_dist(pseudorandom_engine);
+              return static_cast<int>(std::round(boost::math::quantile(skew_dist, probability) * 10));
+            };
+            break;
+          }
+          case DataDistributionType::Pareto: {
+            auto pareto_dist = boost::math::pareto_distribution<double>{column_data_distribution.pareto_scale,
+                                                                        column_data_distribution.pareto_shape};
+            generate_value_by_distribution_type = [pareto_dist, &probability_dist, &pseudorandom_engine]() {
+              const auto probability = probability_dist(pseudorandom_engine);
+              return static_cast<int>(std::floor(boost::math::quantile(pareto_dist, probability)));
```

Comment:
Not sure why you are sometimes using roudn and sometimes floor

Reply:
No idea either. Will change to `round` since this is at least nearer to the actual distribution.